x11rb-protocol/src/protocol/xproto.rs

```diff
@@ -409,6 +551,35 @@ impl Serialize for Visualtype {
         bytes.extend_from_slice(&[0; 4]);
     }
 }
+#[cfg(test)]
+mod visualtype {
+    use super::Visualtype;
+    #[allow(unused_imports)]
+    use crate::x11_utils::{GenRandom, Serialize};
+    use fastrand::Rng;
+    impl GenRandom for Visualtype {
+        fn generate(rng: &Rng) -> Self {
+            Self {
+                visual_id: GenRandom::generate(rng),
+                class: GenRandom::generate(rng),
+                bits_per_rgb_value: GenRandom::generate(rng),
+                colormap_entries: GenRandom::generate(rng),
+                red_mask: GenRandom::generate(rng),
+                green_mask: GenRandom::generate(rng),
+                blue_mask: GenRandom::generate(rng),
+            }
+        }
+    }
+    #[test]
+    fn check_serialize() {
+        let rng = Rng::with_seed(17624827415796400640);
+        let value = Visualtype::generate(&rng);
+        let left = value.serialize();
+        let mut right = alloc::vec![];
+        value.serialize_into(&mut right);
+        assert_eq!(&left[..], right.as_slice());
+    }
```

Comment:
Actually... perhaps this should be run with more than just one seed? "Just to make sure"? Or is that already looking too much like fuzzing?

Also... can this test be moved into a generic function in e.g. `x11_utils`? Something that takes a `&Rng` and is generic over `T: GenRandom + Serialize`? That way, we should end up with less code in total.

Reply:
Yeah, moving it into a generic form is probably for the best, even if it reduces the code amount by only a little.

However, testing with different seeds is unnecessary. Even a difference of 1 will lead to radically different values. If you wanted to we could make it less deterministic by not seeding the generators at all (in which case by default they seed using the thread ID and current system time), but I think we want our tests to be certain.